manifests/init.pp

```diff
@@ -3,24 +3,76 @@
+  $hiera_merge              = undef,
+  $config_dir_name          = undef,
+  $config_dir_purge         = false,
+  $apt_setting              = undef,
+  $apt_setting_hiera_merge  = false,
 ) {
 
-  $hiera_merge_real = str2bool($hiera_merge)
-  validate_bool($hiera_merge_real)
+  $repos_hiera_merge_bool = str2bool($repos_hiera_merge)
+  validate_bool($repos_hiera_merge_bool)
+
+  if $hiera_merge != undef {
+    notify { '*** DEPRECATION WARNING***: Using $hiera_merge is deprecated. Please use $repos_hiera_merge instead!': }
+    $hiera_merge_bool = str2bool($hiera_merge)
+    validate_bool($hiera_merge_bool)
+    if $repos_hiera_merge_bool != $hiera_merge_bool {
+      fail("Different values for \$repos_hiera_merge (${repos_hiera_merge}) and \$hiera_merge (${hiera_merge}). Please use only one.")
+    } else {
+      $repos_hiera_merge_real = $hiera_merge_bool
+    }
+  } else {
+    $repos_hiera_merge_real = $repos_hiera_merge_bool
+  }
+
+  $apt_setting_hiera_merge_real = str2bool($apt_setting_hiera_merge)
+  validate_bool($apt_setting_hiera_merge_real)
+  $config_dir_purge_real = str2bool($config_dir_purge)
+  validate_bool($config_dir_purge_real)
+
+  if $config_dir_name and is_string($config_dir_name) == false {
+    fail('swrepo::config_dir_name is not a string')
+  }
+
   if is_string($repotype) == false { fail('swrepo::repotype is not a string') }
 
+  if $repotype == 'apt' and $::osfamily != 'Debian' { fail('swrepo::repo::repotype with value apt is only valid on osfamily Debian' ) }
+
   case $::osfamily {
-    'RedHat': { $repotype_default = 'yum' }
+    'RedHat': {
+      $repotype_default = 'yum'
+      $config_dir_name_real = '/etc/yum.repos.d'
+      if $config_dir_purge_real == true {
+        file { "${config_dir_name_real}/redhat.repo":
+          require => File[$config_dir_name_real],
+        }
+      }
+    }
     'Suse':   {
+      $config_dir_name_real = '/etc/zypp/repos.d'
       case $::operatingsystemrelease {
         /^(11|12)\./: { $repotype_default = 'zypper' }
-        default:      { fail("Supported osfamilies are RedHat and Suse 11/12. Yours identified as <${::osfamily}-${::operatingsystemrelease}>") }
+        default:      { fail("Supported osfamilies are Debian, RedHat and Suse 11/12. Yours identified as <${::osfamily}-${::operatingsystemrelease}>") }
       }
     }
-    default:  { fail("Supported osfamilies are RedHat and Suse 11/12. Yours identified as <${::osfamily}-${::operatingsystemrelease}>") }
+    'Debian':  {
+      $repotype_default = 'apt'
+      $config_dir_name_real = undef
+    }
+    default: { fail("Supported osfamilies are Debian, RedHat and Suse 11/12. Yours identified as <${::osfamily}-${::operatingsystemrelease}>") }
+  }
+
+  # Manage repo directory
+  if $config_dir_purge_real == true and $config_dir_name_real != undef {
+    file { $config_dir_name_real:
+      ensure  => directory,
+      recurse => $config_dir_purge_real,
+      purge   => $config_dir_purge_real,
```

Comment:
There is no need to use the variables for `recurse` and `purge` parameters as they will be always `true`. Using the variables it is harder to read.